lighthouse-core/gather/gatherers/dobetterweb/links-blocking-first-paint.js

```diff
@@ -21,17 +21,26 @@ const Gatherer = require('../gatherer');
 /* global document,window */
 
 /* istanbul ignore next */
-function getBlockFirstPaintLinks() {
-  // filtered match stylesheet/import
-  // ref)
-  // https://www.igvita.com/2012/06/14/debunking-responsive-css-performance-myths/
-  // https://www.w3.org/TR/html-imports/#dfn-import-async-attribute
-  const linkList = [...document.querySelectorAll('link')].filter(link => {
-    return (link.rel === 'stylesheet' && window.matchMedia(link.media).matches) ||
-           (link.rel === 'import' && link.hasAttribute('async'));
-  }).map(link => link.href);
+function getBlockFirstPaintLinks(filterLinks) {
+  return new Promise((resolve, reject) => {
+    try {
+      const linkList = [...document.querySelectorAll('link')]
+        .filter(filterLinks)
+        .map(link => link.href);
+      resolve(linkList);
+    } catch (e) {
+      reject('Unable to get Stylesheets/HTML Imports on page');
+    }
+  });
+}
 
-  return Promise.resolve(linkList);
+// filtered match stylesheet/import
+// ref)
+// https://www.igvita.com/2012/06/14/debunking-responsive-css-performance-myths/
+// https://www.w3.org/TR/html-imports/#dfn-import-async-attribute
+function filterLinks(link) {
```

Comment:
The refactor wasn't my suggestion, but IMO, it makes the implementation less friendly b/c this function and `getBlockFirstPaintLinks` are actually being injected into the page :\ I won't make you change it back though.